### PR TITLE
Note fading and persistent messages

### PR DESCRIPTION
**Pull request by MrsTonedOne**
Player Notes can now fade out over time so admins don't accidentally ban players for 4 year old notes (admins may still reveal these older notes)

Deleted messages now remain in the database and are instead just hidden from view.

Two things relating to messages I've been meaning to do for ever.
:cl:
add: Player notes can now be configured to fade out over time to allow admins to quickly see how recent notes are. Server Operators, check the config for more info
/:cl:

30 to 90 day fade out preview:

Defaults to disabled on existing servers, defaults to a fade out between 3 months to a year on new servers (What I plan on using for /tg/)

### CODE REVIEW

**Comment by MrStonedOne** [comment]
This will conflict when the other sql prs are merged. I'll just update the version and fix the changelog conflicts when that happens

File: SQL/database_changelog.txt [hunk 1]
```diff
@@ -1,10 +1,25 @@
...
-INSERT INTO `SS13_schema_revision` (`major`, `minor`) VALUES (3, 2);
+INSERT INTO `SS13_schema_revision` (`major`, `minor`) VALUES (3, 3);
+
+----------------------------------------------------
+
+28 August 2017, by MrStonedOne
+Modified table 'messages', adding a deleted column and editing all indexes to include it
+
+ALTER TABLE `messages`
+ADD COLUMN `deleted` tinyint(1) NOT NULL DEFAULT '0' AFTER `edits`,
```
Comment (Jordie0608) [inline SQL/database_changelog.txt]: You missed `unsigned` here.

File: code/controllers/configuration.dm [hunk 2]
```diff
@@ -102,6 +102,8 @@
 	var/use_age_restriction_for_jobs = 0 //Do jobs use account age restrictions? --requires database
 	var/use_account_age_for_jobs = 0	//Uses the time they made the account for the job restriction stuff. New player joining alerts should be unaffected.
 	var/see_own_notes = 0 //Can players see their own admin notes (read-only)? Config option in config.txt
+	var/note_fresh_days
```
Comment (Jordie0608) [inline code/controllers/configuration.dm]: Comments here ploxkthxbye

**Review by Jordie0608 — CHANGES_REQUESTED** [review]
I feel like someone's going to miss hidden notes if they fade entirely; aside from not making them disappear once old enough, what about a counter next to the notes header of total/hidden

i.e.
`Notes (34, 18 Hidden)`
It might get confused with secret/non-secret however, i' not sure what might be better wording.

**Comment by MrStonedOne** [comment]
Jordie, could be some text next to the show all button

**Comment by Jordie0608** [comment]
Obligatory

**Comment by kevinz000** [comment]
>admins not knowing how to read dates

oh come on

File: config/config.txt [hunk 3]
```diff
@@ -241,6 +241,12 @@ TICKLAG 0.5
 ## Uncomment this to let players see their own notes (they can still be set by admins only)
 #SEE_OWN_NOTES
 
+### Comment these two out to prevent notes fading out over time for admins.
+## Notes older then this will start fading out.
+NOTE_FRESH_DAYS 91.31055
```
Comment (lzimann) [inline config/config.txt]: such a strange number on these

File: code/controllers/configuration.dm (hunk 2 above)
Comment (lzimann) [reply code/controllers/configuration.dm]: probably needs a default value aswell

**Comment by Cyberboss** [comment]
You know it's true kev

File: config/config.txt (hunk 3 above)
Comment (MrStonedOne) [reply config/config.txt]: that is 3 avg months.

Comment (lzimann) [reply config/config.txt]: I meant the `.31055` part(same for the next number)

Comment (RemieRichards) [reply config/config.txt]: as he said, it's an average.
a month is often taken to be 4 weeks but this is not consistent, some months have 28 days, others have 31.

Comment (MrStonedOne) [reply config/config.txt]: 1 year = 365.2422 days (to account for leap years)

1 month = 365.2422/12 = 30.43685 days

30.43685 * 3 = 91.31055

**Comment by MrStonedOne** [comment]
I came up with a better solution.

File: code/controllers/configuration.dm (hunk 2 above)
Comment (MrStonedOne) [reply code/controllers/configuration.dm]: nope, the default is set to null intentionally. 0 = start fading out immediately, null = disabled.

**Review by Jordie0608 — APPROVED** [review]

File: SQL/tgstation_schema_prefixed.sql [hunk 4]
```diff
@@ -31,7 +31,7 @@ CREATE TABLE `SS13_admin` (
 -- Table structure for table `SS13_admin_log`
 --
 
-DROP TABLE IF EXISTS `SS13_dmin_log`;
+DROP TABLE IF EXISTS `SS13_admin_log`;
```
Comment (Cyberboss) [inline SQL/tgstation_schema_prefixed.sql]: HA

Comment (Jordie0608) [reply SQL/tgstation_schema_prefixed.sql]: Maybe we should leave it just so nobody tries to use this system.

**Pull request merged**